File: templates/Scripts/hash.py
def hash( value, seed=5 ):
    p = 31
    lp = 1000000007
    ret = seed % lp
    for char in str( value ):
        ret = ( ( ret << 2 ) ^ ( ord( char ) << 1 ) ) % lp
        ret = ( ret >> 1 ) ^ ( p * ret )

    return hex ( abs( ret ) )


apis = [
    "CreateProcessA",
    "VirtualAllocEx",
    "WriteProcessMemory",
    "QueueUserAPC",
    "ResumeThread",
    "WaitForSingleObject",
    "CloseHandle",
    "VirtualFreeEx",
    "1000",
]

for api in apis:
    print(f"{ api } -> { hash( api ) }")

File: templates/Source/Hash.c
#include <stdint.h>
#include <stdio.h>
#include <windows.h>

uint64_t Hash( const char * str, uint64_t seed ) {
    const uint64_t p  = 31;
    const uint64_t lp = 1000000007;
    uint64_t ret      = seed % lp;

    while ( *str ) {
        ret = ( ( ret << 2 ) ^ ( (uint64_t)( *str ) << 1 ) ) % lp;
        ret = ( ret >> 1 ) ^ ( p * ret );
        str++;
    }
    return ret;
}